cmake/edge_configure.cmake

```diff
@@ -109,9 +109,23 @@ if (${FIRMWARE_UPDATE})
   MESSAGE ("Enabling firmware update for Edge")
 
   if (FOTA_ENABLE)
-    execute_process ( COMMAND cp ${CMAKE_CURRENT_SOURCE_DIR}/fota/fota_install_callback.c ${CMAKE_CURRENT_SOURCE_DIR}/lib/mbed-cloud-client/fota/ )
+    execute_process ( COMMAND cp ${CMAKE_CURRENT_SOURCE_DIR}/fota/fota_app_callbacks.c ${CMAKE_CURRENT_SOURCE_DIR}/lib/mbed-cloud-client/fota/ )
+    if (NOT DEFINED FOTA_SCRIPT_DIR)
+      SET (FOTA_SCRIPT_DIR \"/opt/pelion\")
+    endif()
+    if (NOT DEFINED FOTA_INSTALL_MAIN_SCRIPT)
+      SET (FOTA_SCRIPT_DIR \"fota_update_activate.sh\")
+    endif()
+
     add_definitions(-DFOTA_DEFAULT_APP_IFS=1)
     add_definitions(-DTARGET_LIKE_LINUX=1)
+    add_definitions(-DFOTA_SCRIPT_DIR=${FOTA_SCRIPT_DIR})
+    add_definitions(-DFOTA_INSTALL_MAIN_SCRIPT=${FOTA_INSTALL_MAIN_SCRIPT})
+    if (FOTA_COMBINED_IMAGE_SUPPORT) 
+      add_definitions(-DMBED_CLOUD_CLIENT_FOTA_SUB_COMPONENT_SUPPORT=1)
+      add_definitions(-DFOTA_CUSTOM_PLATFORM=1)
```

Comment:
@davidsaada what is the purpose of this flag?

Reply:
The way FOTA works, any platform specific hooks required in cases of init or update start/finish/abort require this flag to be set. In our case, we need to have `fota_platform_init_hook` to create the combined update handlers.